cmake/TorchConfig.cmake.in

```diff
@@ -133,11 +136,16 @@ if(@USE_CUDA@)
 
   if(@BUILD_SHARED_LIBS@)
     find_library(C10_CUDA_LIBRARY c10_cuda PATHS "${TORCH_INSTALL_PREFIX}/lib")
-    list(APPEND TORCH_CUDA_LIBRARIES ${C10_CUDA_LIBRARY} ${Caffe2_PUBLIC_CUDA_DEPENDENCY_LIBS})
+    find_library(TORCH_CUDA_LIBRARY c10_cuda PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    list(APPEND TORCH_CUDA_LIBRARIES ${C10_CUDA_LIBRARY} ${TORCH_CUDA_LIBRARY} ${Caffe2_PUBLIC_CUDA_DEPENDENCY_LIBS})
```

Comment:
To keep `TORCH_CUDA_LIBRARIES` BC, here doesn't directly use `append_torchlib_if_found`. Because `append_torchlib_if_found` doesn't update `TORCH_CUDA_LIBRARIES`.

Reply:
This change seems unrelated to the PR title. Do you mind proposing it in separate PR (and posting some sort of examples of what behavior it fixes)